Documentation/ApiOverview/Backend/BackendModules/InitializeModuleTemplate.rst.txt

```diff
@@ -0,0 +1,32 @@
+..  Generated by https://github.com/TYPO3-Documentation/t3docs-codesnippets
+..  Extracted from T3docs\BlogExample\Controller\BackendController
+
+..  code-block:: php
+    :caption: Class T3docs\\BlogExample\\Controller\\BackendController
+
+    use Psr\Http\Message\ResponseInterface;
+    use Psr\Http\Message\ServerRequestInterface;
+    use T3docs\BlogExample\Domain\Model\Post;
+    use TYPO3\CMS\Backend\Template\ModuleTemplate;
+
+    class BackendController extends ActionController
+    {
+        protected function initializeModuleTemplate(
+            ServerRequestInterface $request,
+        ): ModuleTemplate {
+            $view = $this->moduleTemplateFactory->create($request);
+
+            $this->modifyDocHeaderComponent($view);
+            $view->setFlashMessageQueue($this->getFlashMessageQueue());
+
+            return $view;
+        }
+
+        public function showPostAction(Post $post): ResponseInterface
+        {
+            $view = $this->initializeModuleTemplate($this->request);
+            $view->assign('post', $post);
+            return $view->renderResponse('ShowPost');
```

Comment:
The request was added to the `$view`, so it knows which template has to be loaded. IMO you can leave that parameter empty to load default template for that action.

Reply:
This code is automatically generated from the blog example. You can provide a PR there and it will regenerate here with the next regeneration of the code snippets